File: Backend/models/recipe_comment.py
from database.db import db
from datetime import datetime

class RecipeComment(db.Model):
    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey("users.id"), nullable=False)
    recipe_id = db.Column(db.Integer, db.ForeignKey("recipes.id"), nullable=False)
    title = db.Column(db.String(255), nullable=False)
    content = db.Column(db.Text, nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    image = db.Column(db.String(255), nullable=True)

    user = db.relationship("User", back_populates="comments")
    recipe = db.relationship("Recipe", back_populates="comments")

File: Frontend/src/lib/types/RecipeComment.ts
import type { Recipe } from "./Recipe";
import type { User } from "./User";

export interface RecipeComment{
    id: number;
    recipe_id: number;
    user: {
        user_id: number,
        email: string
    };

    title: string;
    content: string;
    image?: string;
}